docs/development/design/exit-node.md

```diff
@@ -0,0 +1,157 @@
+# Exit Node Support
+
+## Summary
+
+This document proposes exit node functionality in Nexodus. An exit node in the context of Nexodus is where traffic from a device will drain to a designated Nexodus agent via a peering and default route associated with the Nexodus exit node.
+
+## Proposal
+
+The initial scope of work is for Linux only. There are two main components, the exit node which originates all traffic leaving the Nexodus mesh and the exit node client which subscribes to the exit node service.
+
+### Exit Node
+
+- An exit node is a node that is advertising the network address `0.0.0.0/0`, the default route CIDR notation. Other devices in the organization are able to identify an exit node based on it advertising a default route prefix. The initial implementation is not adding any database fields to the API server schema.
+- One additional flag is required to make sure a user intends on originating a default route that will forward traffic from a nexd driver interface such as `wg0` to another port such as `eth0`. The `--exit-node` flag also informs the agent to perform source NAT for traffic leaving the Nexodus mesh to allow return traffic to return to the originating exit-node and then sent back to the pre-nat source device.
+- To begin with, there will be at most one active exit node per organization (or once a VPC/slice capability is implemented). Looking forward, the exit-node client should have the ability to choose from multiple exit nodes in an organization or slice of an organization and choose from multiple exit nodes based on the user's criteria, such as latency or geolocation. Since these are default routes that overlap being advertised to the mesh, the driver implementation will dictate whether more than one exit node can be concurrently active. In the case of the Wireguard driver, a peer's public key is associated to the route, so it is not possible to overlap prefixes.
+- The exit node startup would look like the following example:
+
+```shell
+nexd router --exit-node
+```
+
+### Exit Node Client
+
+- In order for a client to use the exit node, the user will need to explicitly subscribe to the exit node. Since exit nodes can be disruptive to the user's connection by tunneling all traffic to the exit node, it makes sense to have the extra step needed to explicitly join an exit node. The user can join via `nexctl` or via the web UI.
```

Comment:
```suggestion
- In order for a client to use the exit node, the user will need to explicitly subscribe to the exit node. Since exit nodes can be disruptive to the user's connection by tunneling all traffic to the exit node, it makes sense to have the extra step needed to explicitly join an exit node. The user can join via `nexctl`, via the web UI, or via a command-line argument to `nexd`.
```